Fix/clean lightbox issue

zihanKuang opened the pull request:
**Notes for Reviewers**  

- This PR improves image handling in the documentation by enabling Lightbox support for SVGs and providing clear guidelines for consistent image usage.  


- [x] Yes, I signed my commits.  

---

### **Problem Statement**  

Currently, the documentation contains **three different image usage patterns**, leading to inconsistency in behavior and formatting:  

1. **Links opening in a new tab (full-page view)**  
```markdown
[![Context Aware Design]({{ site.baseurl }}/assets/img/patterns/context-aware-design.svg
)]({{ site.baseurl }}/assets/img/patterns/context-aware-design.svg)
```  
- Clicking opens the image in a **new browser tab** instead of a Lightbox.  
- **SVGs were not supported in Lightbox**, causing an inconsistent experience.  

2. **Static inline images (non-clickable, fixed size)**  
```html
<img src="{{ site.baseurl }}/assets/img/configuration-management/meshery-patterns.png" width="50%" />
```  
- These images **do not support enlargement** when clicked.  
- Used in **~50+ locations** across various doc pages.  

3. **Lightbox-enabled images (clickable, overlay effect)**  
```html
<a href="{{ site.baseurl }}/assets/img/notification-center/NotificationCenterOverview.png">
    <img alt="Notification Center Overview" style="width:500px;height:auto;" 
         src="{{ site.baseurl }}/assets/img/notification-center/NotificationCenterOverview.png" />
</a>
```  
- Clicking opens a **Lightbox overlay** for better viewing.  

---

### **Solution Implemented**  

####  **1. Enabling Lightbox for SVGs**  
 **Fix** – Modified the following files to support SVGs in Lightbox:  
- **`assets/css/lightbox.css`**  
- **`assets/js/lightbox.js`**  

 **Result** – Existing Markdown-style image references (Pattern 1) now **work with Lightbox** **without requiring any code changes**.  

####  **2. Handling Static Images (Pattern 2)**  
 **Observations**:  
- Around **50+ images** use the static `<img>` format.  
- They are **mixed with UI elements (logos, icons)**, making batch updates difficult.  

 **Proposal**:  
- These static images can be **updated incrementally** when modifying related docs.  
- Some images **do not need Lightbox** (e.g., large, clear images).  

---

### **Contributor Guide Update** (Standardized Image Usage)  

To ensure consistency, the documentation now provides **clear guidelines**:

l5io commented:
🚀 Preview for commit 4d9795eccf67a63c4803840dbb61136d7d524132 at: https://67bb16b03ebf770b1ce8a87a--meshery-docs-preview.netlify.app

l5io commented:
🚀 Preview for commit 50de59e4caa8fea63f31568136bbbdd9ca9290d9 at: https://67bb18924b59d70be8203a46--meshery-docs-preview.netlify.app

vishalvivekm commented:
@zihanKuang 
Thank you for your contribution! Let's discuss this during the website call on Today at 7 AM CT.
adding it as an agenda item to the [meeting minutes](https://docs.google.com/document/d/1XczAHXVe2FIWPqiF57ospJ43zw5cZQ7ui8mn39v5EvA/edit?usp=drivesdk).

zihanKuang commented:
Hi @vishalvivekm
Thank you for the update! I have class in this time, so I won’t be able to join the meeting today. However, I’m available for a call on Tuesday or Wednesday😃

leecalcote commented:
@GARY121github would you like to offer feedback and review of this PR?

marblom007 reviewed (dismissed):
✅

marblom007 commented:
Is there any remaining work to be done here?

leecalcote commented:
@zihanKuang signal when ready for a final review. 👍

zihanKuang commented:
Hey @leecalcote 
I've completed the updates and addressed the necessary changes. And ready for a final review. 👍

Also, @marblom007, I really appreciate your previous review!  Since I made some additional updates, GitHub automatically dismissed the stale review😥. I updated the **Effect** section in contributing-docs.md to provide a clearer explanation. Feel free to take another look if needed—thanks again for your time!

l5io commented:
🚀 Preview for commit f752459c3286a84318e2ae776183cd53b574d76c at: https://67bfa159bc749e6f31c2085f--meshery-docs-preview.netlify.app

marblom007 commented:
It’s a pleasure to review your PR given the level of diligence you have performed here.Its description is very well written.

leecalcote commented on docs/pages/project/contributing/contributing-docs.md:
```diff
@@ -509,6 +509,44 @@ To use the `alert.html` feature in our documentation include the following code:
+- Image width is fixed at `500px`, maintaining aspect ratio
+- If `500px` exceeds `90vw`, it will be constrained to `90vw`
+- Clickable, opening in Lightbox for full-size viewing
+
```
It’s very good to see this documentation included.
